### PR TITLE
Add Apache Flink release 1.10.1

### PR DESCRIPTION
**Pull request by carp84**
Adds release announcement and download link for 1.10.1
Dates on announcements are still TBD and should be updated accordingly once release happens.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -194,8 +194,8 @@ release_archive:
     flink:
       -
         version_short: "1.10"
-        version_long: 1.10.0
-        release_date: 2020-02-11
+        version_long: 1.10.1
```
Comment (hequn8128) [inline _config.yml:197]: Add 1.10.1 instead of replacing 1.10.0.

File: _posts/2020-05-05-release-1.10.1.md [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+categories: news
+authors:
+- liyu:
+  name: "Yu Li"
+  twitter: "LiyuApache"
+---
+
+The Apache Flink community released the first bugfix version of the Apache Flink 1.10 series.
+
+This release includes 143 fixes and minor improvements for Flink 1.10.0. The list below includes a detailed list of all fixes and improvements.
```
Comment (hequn8128) [inline _posts/2020-05-05-release-1.10.1.md]: 143 => 152

**Review by hequn8128 — COMMENTED** [review]
@carp84 Thanks a lot for managing the release! Left some minor comments.

File: _posts/2020-05-05-release-1.10.1.md (hunk 2 above)
Comment (carp84) [reply _posts/2020-05-05-release-1.10.1.md]: good catch.

**Comment by carp84** [comment]
Thanks for the review @hequn8128 . All comments addressed, PTAL. Thanks.

File: _config.yml [hunk 3]
```diff
@@ -192,6 +192,10 @@ component_releases:
 
 release_archive:
     flink:
+      -
+        version_short: "1.10"
```
Comment (hequn8128) [inline _config.yml:196]: This is right but it would be better to make it consistent with others, i.e., remove the quotes.

**Review by hequn8128 — APPROVED** [review]
@carp84 Thanks a lot for the update. LGTM.

File: _config.yml (hunk 3 above)
Comment (carp84) [reply _config.yml:196]: Thanks for the review. The quotes are for preventing issues like reported in FLINK-16663, FYI.

**Review by rmetzger — APPROVED** [review]
LGTM

**Comment by dianfu** [comment]
@carp84 Thanks a lot for managing this release. I have corrected the fix version of the following JIRAs:
1) Marked the fix version of FLINK-17499, FLINK-16944, FLINK-16225, FLINK-16014, FLINK-15741, FLINK-15143 and FLINK-15933 as 1.10.1
2) Removed the fix version of "1.10.1" for FLINK-15998, FLINK-11072, FLINK-14223, FLINK-16142 and FLINK-15593 as they are not fixed in 1.10.1

We need to update this website PR to reflect these changes before merging.

**Review by uce — APPROVED** [review]
LGTM.

**Comment by carp84** [comment]
Thanks @dianfu . I think most edits of the JIRAs make sense but have some concerns around FLINK-16225. It's still not fixed although already one commit goes into release-1.10, and IMO we shouldn't have any open issues included into our release note. Wdyt?

**Comment by dianfu** [comment]
@carp84 Thanks for the double check. You are right and have marked it back to 1.10.2.

**Review by dianfu — APPROVED** [review]

**Comment by sjwiesman** [comment]
@carp84 We have discovered a breaking binary compatibility change in the StreamingFileSink 1.101. Based on the discussion on the dev ML I believe we should add a notice to the release notes. 

https://lists.apache.org/thread.html/r293768d13d08149d756e0bf91be52372edb444c317535d1d5a496c3e%40%3Cdev.flink.apache.org%3E

My proposal: 

> After release the community discovered a binary incompatible change was merged into Flink 1.10.1. [FLINK-16684](https://issues.apache.org/jira/browse/FLINK-16684) changed the builders of the StreamingFileSink to make them compilable in Scala. This change is source compatible but binary incompatible. If using the StreamingFileSink, please recompile your user code against 1.10.1 before upgrading.

**Comment by tweise** [comment]
+1 for the release note addition for the binary incompatible change.

**Comment by carp84** [comment]
Release note updated, adding note for:
1. The binary incompatible change of FLINK-16684
2. We don't support starting cluster with .bat scripts on Windows anymore (FLINK-16683)

The preview looks like:

image

Please let me know if any more comments, thanks.

**Review by tweise — APPROVED** [review]

**Comment by carp84** [comment]
PR merged, thanks all for review!

**Pull request merged**